spec/models/state_changers_spec.rb

```diff
@@ -125,6 +125,24 @@
           expect(work_completion_creation).to_not have_been_made
         end
       end
+
+      # The ability to have multiple request types in the config was added for scRNA Core pipeline.
+      # The expectation was that any one plate would only have one of the request types on it,
+      # so I haven't tested a plate with a mix of request types.
+      context 'when one of the multiple config request types matches the in progress submissions' do
```

Comment:
It occurs to me that aggregation is like cherrypicking. So they may return to the same source and pick a different subset of wells (or even repeat pick some). So multiple requests_as_source in a well is possible. Only one should be active at a time though.
I think this is fine for work completion, as I think the filtering for active requests is happening on the SS side.
But for the de-duplication code further up, do we need to consider filtering requests_as_source for active requests?

Reply:
what is the `requests_for_well` method used for again?

Reply:
`requestsForWell` is used in `requestsFromPlates` in `requestHelpers.js`

That's then used in `MultiStamp.vue`, where it then calls `requestIsActive` on them.

So `requestsForWell` simply grabs the requests, and they're filtered later.

Reply:
requests_as_source implies transfer requests on receptacle I think.